node_examples/infinite_stream.js

```diff
@@ -0,0 +1,221 @@
+            break;
+    }
+}
+
+/**
+ * Start revai client stream and set up handlers
+ */
+function startStream() {
+    client = new revai.RevAiStreamingClient(token, audioConfig);
+
+    // Create your event responses
+    client.on('close', (code, reason) => {
+        console.log(`Connection closed, ${code}: ${reason}`);
+        if (code !== 1000){
+            console.log("restarting stream");
+            restartStream();
+        }
+        console.log(bytesWritten);
+    });
+    client.on('httpResponse', code => {
+        console.log(`Streaming client received http response with code: ${code}`);
+    });
+    client.on('connectFailed', error => {
+        console.log(`Connection failed with error: ${error}`);
+    });
+    client.on('connect', connectionMessage => {
+        console.log(`Connected with job id: ${connectionMessage.id}`);
+        
+        // Trigger a write in order to restart sending data if source has ended
+        audioInputStreamTransform.write(Buffer.alloc(0));
+    });
+    
+    audioBackup = [];
+    sessionConfig.startTs = lastResultEndTsReceived;
+    
+    revaiStream = client.start(sessionConfig);
+    revaiStream.on('data', data => {
+        handleData(data);
+    });
+    revaiStream.on('error', error => {
+        console.log(error);
+    })
+    revaiStream.on('end', function () {
+        console.log('End of Stream');
+    });
+}
+
+let bytesWritten = 0;
+
+/**
+ * Write audio to revai websocket. Handles storing audio copy and resending from it if needed
+ */
+const audioInputStreamTransform = new Writable({
+    write(chunk, encoding, next) {
+        if (newStream && audioBackupCopy.length !== 0) {
+            // Calculate audio to resend based on timestamp of last hypothesis received
+            const bitsSent = lastResultEndTsReceived * samplesPerSecond * bytesPerSample;
+            const chunksSent = Math.floor(bitsSent / chunkSize);
+            if (audioBackupCopy.length - chunksSent !== 0) {
+                for (let i = chunksSent; i < audioBackupCopy.length; i++) {
+                    revaiStream.write(audioBackupCopy[i][0], audioBackupCopy[i][1]);
+                }
+            }
+            newStream = false;
+        }
+
+        audioBackup.push([chunk, encoding]);
+
+        if (revaiStream) {
+            revaiStream.write(chunk, encoding);
+            bytesWritten += chunk.length;
+        }
+
+        next();
+    },
+
+    final() {
+        if (client && revaiStream) {
+            client.end();
+            revaiStream.end();
+        }
+    }
+});
+
+/**
+ * Restart the stream
+ */
+function restartStream() {
+    if (revaiStream) {
+        revaiStream.removeListener('data', handleData);
+        revaiStream = null;
+    }
+
+    audioBackupCopy = [];
+    audioBackupCopy = audioBackup;
+
+    newStream = true;
+
+    startStream();
+}
+
+// Read file from disk
+let file = fs.createReadStream('./resources/example.raw');
```

Comment:
I think the file path should be a const at the top and next to the `audioConfig`. Most people using this for real will probably want to swap the audio out